src/commands/run.rs

```diff
@@ -0,0 +1,263 @@
+    #[arg(short, long, value_name = "NAME", conflicts_with = "inputs")]
+    pub name: Option<String>,
+
+    /// The output directory; defaults to the task name.
+    ///
+    /// If no output directory is provided, a default nested directory is
+    /// created based on the task name and the current time in the form
+    /// `sprocket_runs/<task_name>/<timestamp>/`.
+    #[arg(short, long, value_name = "DIR")]
+    pub output: Option<PathBuf>,
+
+    /// Overwrite the output directory if it exists.
+    #[arg(long)]
+    pub overwrite: bool,
+
+    /// Disables color output.
+    #[arg(long)]
+    pub no_color: bool,
+
+    /// The report mode.
+    #[arg(short = 'm', long, default_value_t, value_name = "MODE")]
+    pub report_mode: Mode,
+}
+
+/// Creates the output directory for the task.
+fn create_output_dir(output_dir: Option<PathBuf>, name: &str, overwrite: bool) -> Result<PathBuf> {
+    let output_dir = output_dir.unwrap_or_else(|| {
+        let timestamp = chrono::Utc::now().format("%Y%m%d_%H-%M-%S");
+        PathBuf::from(format!("sprocket_runs/{}/{}", name, timestamp))
+    });
+
+    let output_dir = if output_dir.exists() {
+        if overwrite {
+            fs::remove_dir_all(&output_dir).with_context(|| {
+                format!(
+                    "failed to remove output directory `{dir}`",
+                    dir = output_dir.display()
+                )
+            })?;
+            output_dir
+        } else {
+            bail!(
+                "output directory `{dir}` already exists; use the `--overwrite` option to \
+                 overwrite it",
+                dir = output_dir.display()
+            );
+        }
+    } else {
+        output_dir
+    };
+
+    fs::create_dir_all(&output_dir).with_context(|| {
+        format!(
+            "failed to create output directory `{dir}`",
+            dir = output_dir.display()
+        )
+    })?;
+
+    log::info!(
+        "output directory created: `{dir}`",
+        dir = output_dir.display()
+    );
+
+    Ok(output_dir)
+}
+
+/// Runs a task.
+pub async fn run(args: RunArgs) -> Result<()> {
+    eprintln!(
+        "the `run` command is in alpha testing and does not currently support workflows or using \
+         containers."
+    );
+
+    let file = args.file;
+    if Path::new(&file).is_dir() {
+        anyhow::bail!("expected a WDL document, found a directory");
+    }
+    let (config, mut stream) = get_display_config(args.report_mode, args.no_color);
+
+    let results = analyze(&file, vec![], false, false).await?;
+
+    let uri = if let Ok(uri) = Url::parse(&file) {
+        uri
+    } else {
+        path_to_uri(&file).expect("file should be a local path")
+    };
+
+    let result = results
+        .iter()
+        .find(|r| **r.document().uri() == uri)
+        .context("failed to find document in analysis results")?;
+    let document = result.document();
+
+    let mut engine = Engine::new(LocalTaskExecutionBackend::new());
```

Comment:
Do we have a concept for how the execution backend will be chosen in the future?

Reply:
I don't have much of a vision for that yet, but I imagine Clay and Peter have thoughts. I think it's safe to assume that without advanced configuration/options, we will default to the local backend.